3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -18,28 +18,34 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
 function requestData(url) {
-  // TODO return a promise using `fetch()`
+  return fetch(url).then((response) => {
+    if (response.ok) {
+      return response.json();
+    } else {
+      throw new Error('Request failed!');
+    }
```

Comment:
```suggestion
    if (response.ok) {
      return response.json();
    }
    throw new Error('Request failed!');
```

... well, it would be more concise if you left out the unnecessary `else`. After the `return` there is no need for it.